tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/environment/DockerComposeExecutor.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized.environment;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+import org.hawkular.apm.tests.dockerized.exception.EnvironmentException;
+import org.hawkular.apm.tests.dockerized.model.TestEnvironment;
+
+/**
+ * @author Pavol Loffay
+ */
+public class DockerComposeExecutor implements TestEnvironmentExecutor {
+    private static final String HAWKULAR_APM_NETWORK = "hawkular-apm";
+
+    private final String apmBindAddress;
+
+
+    /**
+     * @param apmBindAddress Address of default gateway (host OS where is running APM server). This network
+     *                       is also used as default in docker-compose
+     */
+    public DockerComposeExecutor(String apmBindAddress) {
+        this.apmBindAddress = apmBindAddress;
+    }
+
+    /**
+     * Create network which is used as default in docker-compose.yml
+     * This should be run before {@link DockerComposeExecutor#run(TestEnvironment)}
+     */
+    public void createNetwork() {
+        String[] command = new String[] {
+                "docker", "network", "create", "--driver=bridge", "--subnet=" + apmBindAddress + "/24",
+                "--gateway=" + apmBindAddress, HAWKULAR_APM_NETWORK,
+        };
+
+        runShellCommand(command);
```

Comment:
I can use docker java client for this, however the one I'm using (spotify) does not have this implemented. I can change to another.